Apps/WebClient/src/NewClientApp/src/stores/waitlist.ts

```diff
@@ -0,0 +1,208 @@
+            tooBusyField.value = false;
+            status.value = LoadStatus.LOADED;
+        }
+
+        function clearTicket() {
+            clearTimeout(checkInTimeoutId.value);
+            ticketField.value = undefined;
+            tooBusyField.value = false;
+            status.value = LoadStatus.NONE;
+        }
+
+        function setCheckInTimeoutId(incomingCheckInTimeoutId: number) {
+            checkInTimeoutId.value = incomingCheckInTimeoutId;
+        }
+
+        function scheduleCheckIn(
+            ticket?: Ticket,
+            getTicketOnFail: boolean = false
+        ): void {
+            clearTimeout(checkInTimeoutId.value);
+            const localTicket = ticket ?? ticketField.value;
+            if (localTicket === undefined) {
+                logger.debug(`Waitlist - Ticket undefined`);
+                return;
+            }
+            const now = new Date().getTime();
+            const checkInAfter = localTicket.checkInAfter * 1000;
+            const timeout = Math.max(0, checkInAfter - now);
+            logger.debug(`Waitlist - Scheduling checkIn in ${timeout}ms`);
+            const newTimeoutId = window.setTimeout(() => {
+                checkIn().catch(() => {
+                    logger.error(`Waitlist - Scheduled checkIn failed`);
+                    if (getTicketOnFail) {
+                        getTicket().catch(() =>
+                            logger.error(
+                                `Waitlist - Scheduled checkIn failed to get ticket`
+                            )
+                        );
+                    }
+                });
+            }, timeout);
+            if (newTimeoutId !== undefined) {
+                setCheckInTimeoutId(newTimeoutId);
+            } else {
+                logger.error(
+                    `Waitlist - Scheduled checkIn failed to set timeout`
+                );
+            }
+        }
+
+        function getTicket(): Promise<Ticket> {
+            const localTraceId = "error-ticket-undefined";
+            logger.debug("Waitlist - Retrieving ticket");
+            setTicketRequested();
+            return ticketService
+                .createTicket("healthgateway")
+                .then((result) => {
+                    if (result) {
+                        setTicket(result);
+                        handleTicket(result);
+                        return result;
+                    } else {
+                        setError();
+                        throw {
+                            resultMessage: "Ticket is undefined",
+                            traceId: localTraceId,
+                        } as ResultError;
+                    }
+                })
+                .catch((error: ResultError) => {
+                    if (error.traceId === localTraceId) {
+                        setError();
+                    } else {
+                        setTooBusy();
+                    }
+                    throw error;
```

Comment:
When no ticket is returned, this is going to call `setError()` twice (and it won't call `setTooBusy()`, but I think that might be fine).